### PR TITLE
Adds a couple of test cases

### PR DESCRIPTION
**Pull request by jamesdavidson**
Specifically, for the honour_origin_cache_headers custom matcher.

### CODE REVIEW

**Comment by biancaG** [comment]
Thanks, it looks good. I'm not sure the comments before each test are necessary, given that we have the description. Should any extra information in the comments go in to the description?

**Comment by jamesdavidson** [comment]
Yeah fair call, the rspec syntax is quite descriptive after all.

File: spec/functional/honour_origin_cache_headers_spec.rb [hunk 1]
```diff
@@ -1,9 +1,66 @@
 describe 'honour_origin_cache_headers' do
+
+  let(:a_date_in_the_future) { 'Thu, 01 Dec 2015 07:00:00 GMT' }
+  let(:a_date_in_the_future_plus_one) { 'Thu, 01 Dec 2015 07:01:00 GMT' }
```
Comment (biancaG) [inline spec/functional/honour_origin_cache_headers_spec.rb]: Good job giving descriptive names :)

Comment (jamesdavidson) [inline spec/functional/honour_origin_cache_headers_spec.rb]: Though one problem is that on Dec 1st it'll probably break!

Comment (biancaG) [inline spec/functional/honour_origin_cache_headers_spec.rb]: Oh, of course! How about we change it to be relative to the current date, or use timecop?

Comment (jamesdavidson) [inline spec/functional/honour_origin_cache_headers_spec.rb]: I did try making it relative but I got stuck trying to extract the right format string from Ruby's Time API. I had previously been trying to find the correct answer in the RFC documents (822, 2616 and 1123) but just got myself confused. However! Looking at the problem again, I see the method we need here is just Time#httpdate.

This should do the trick: 
let(:a_date_in_the_future) { (Time.now + 6*30*24*3600).httpdate }
let(:a_date_in_the_future_plus_one) { (Time.now + 6*30*24*3600 + 1).httpdate }

Comment (biancaG) [inline spec/functional/honour_origin_cache_headers_spec.rb]: Hmm, I think timecop and freezing would be cleaner...

Comment (jamesdavidson) [inline spec/functional/honour_origin_cache_headers_spec.rb]: Perhaps, up to you. To be clear, you are proposing freezing at some fixed timestamp and testing against some other fixed timestamp?

Comment (jamesdavidson) [inline spec/functional/honour_origin_cache_headers_spec.rb]: Interestingly, if you change the current date from 2015 to 2014... the test actually just passes, no problem!

**Pull request merged**